modules/codegen/src/test/scala/core/issues/Issue314.scala

```diff
@@ -0,0 +1,165 @@
+       |    val allHeaders = headers ++ scala.collection.immutable.Seq[Option[HttpHeader]]().flatten
+       |    makeRequest(HttpMethods.GET, host + basePath + "/user/" + Formatter.addPath(id), allHeaders, HttpEntity.Empty, HttpProtocols.`HTTP/1.1`).flatMap(req => EitherT(httpClient(req).flatMap(resp => resp.status match {
+       |      case StatusCodes.OK =>
+       |        resp.discardEntityBytes().future.map(_ => Right(GetUserResponse.OK))
+       |      case _ =>
+       |        FastFuture.successful(Left(Right(resp)))
+       |    }).recover({
+       |      case e: Throwable =>
+       |        Left(Left(e))
+       |    })))
+       |  }
+       |}""".stripMargin
+
+  private def companion(prefix: String) =
+    s"""object ${prefix}Client {
+       |  def apply(host: String = "http://localhost:1234")(implicit httpClient: HttpRequest => Future[HttpResponse], ec: ExecutionContext, mat: Materializer): ${prefix}Client = new ${prefix}Client(host = host)(httpClient = httpClient, ec = ec, mat = mat)
+       |  def httpClient(httpClient: HttpRequest => Future[HttpResponse], host: String = "http://localhost:1234")(implicit ec: ExecutionContext, mat: Materializer): ${prefix}Client = new ${prefix}Client(host = host)(httpClient = httpClient, ec = ec, mat = mat)
+       |}""".stripMargin
+
+  private def handler(prefix: String) =
+    s"trait ${prefix}Handler { def getUser(respond: ${prefix}Resource.GetUserResponse.type)(id: String): scala.concurrent.Future[${prefix}Resource.GetUserResponse] }"
+
+  private def resource(prefix: String) =
+    s"""object ${prefix}Resource {
+       |  def routes(handler: ${prefix}Handler)(implicit mat: akka.stream.Materializer): Route = {
+       |    {
+       |      path("user" / Segment).apply(id => get(discardEntity(complete(handler.getUser(GetUserResponse)(id)))))
+       |    }
+       |  }
+       |  sealed abstract class GetUserResponse(val statusCode: StatusCode)
+       |  case object GetUserResponseOK extends GetUserResponse(StatusCodes.OK)
+       |  object GetUserResponse {
+       |    implicit val getUserResponseTRM: ToResponseMarshaller[GetUserResponse] = Marshaller { implicit ec =>
+       |      resp => getUserResponseTR(resp)
+       |    }
+       |    implicit def getUserResponseTR(value: GetUserResponse)(implicit ec: scala.concurrent.ExecutionContext): scala.concurrent.Future[List[Marshalling[HttpResponse]]] = value match {
+       |      case r: GetUserResponseOK.type =>
+       |        scala.concurrent.Future.successful(Marshalling.Opaque {
+       |          () => HttpResponse(r.statusCode)
+       |        } :: Nil)
+       |    }
+       |    def apply[T](value: T)(implicit ev: T => GetUserResponse): GetUserResponse = ev(value)
+       |    def OK: GetUserResponse = GetUserResponseOK
+       |  }
+       |}""".stripMargin
+
+  private def verify(config: String, expectedClassPrefix: String): Unit = {
```

Comment:
Make sure to add the implicit `loc` (or `pos` or whatever it's called) to let scalatest know which line of the test failed, otherwise it'll just give the line number of the `should` at the bottom of all these `verify` methods